components/director/pkg/graphql/query_params.go

```diff
@@ -28,3 +29,25 @@ func (y QueryParams) MarshalGQL(w io.Writer) {
 		return
 	}
 }
+
+type QueryParamsSerialized string
+
+func (y *QueryParamsSerialized) Unmarshal() (map[string][]string, error) {
+	var data map[string][]string
+	if y == nil {
+		return data, nil
+	}
+
+	err := json.Unmarshal([]byte(*y), &data)
+
+	return data, err
```

Comment:
I think it would be nice to add one of our custom errors here, eg. `InvalidData`. What do you think about that?